Usecase/usecase_transaction_test.go

```diff
@@ -75,8 +75,8 @@ func TestTransactionUseCase_CreateTransaction(t *testing.T) {
 	})
 
 	t.Run("Cannot create transaction", func(t *testing.T) {
-		mockProductRepo.On("GetPriceProducts", transaction).Return(transaction, nil).Once()
-		mockTransactionRepo.On("SaveCreateTransaction", transaction).Return(nil, errors.New("cannot create transaction")).Once()
+		mockProductRepo.On("GetPriceProducts", &transaction).Return(&transaction, nil).Once()
```

Comment:
Why **GetPriceProducts()** and other function in repository are need  input with address value (**&**transaction)

Reply:
<img width="805" alt="Screenshot 2567-07-15 at 14 52 51" src="https://github.com/user-attachments/assets/b44083e8-9159-47c5-9afd-de96a4fef1a2">

1. because this function update transaction itself and return transaction updated value, So parameter use pointer variable, I think so.